src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/Grid.cs

```diff
@@ -272,30 +273,70 @@ public bool ShowGridLines
         /// <summary>
         /// Returns a ColumnDefinitionCollection of column definitions.
         /// </summary>
+        [TypeConverter(typeof(ColumnDefinitionCollectionConverter))]
         [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
         public ColumnDefinitionCollection ColumnDefinitions
         {
             get
             {
-                if (_data == null) { _data = new ExtendedData(); }
-                if (_data.ColumnDefinitions == null) { _data.ColumnDefinitions = new ColumnDefinitionCollection(this); }
-
-                return (_data.ColumnDefinitions);
+                if (_data == null) _data = new ExtendedData();
+                if (_data.ColumnDefinitions == null)
+                {
+                    _data.ColumnDefinitions = new ColumnDefinitionCollection(this);
+                }
+                return _data.ColumnDefinitions;
+            }
+            set
+            {
+                if (value == null){
+                    _data.ColumnDefinitions = new ColumnDefinitionCollection(this);
+                    return;
+                }
+                if (_data == null) _data = new ExtendedData();
+                if (_data.ColumnDefinitions == null)
+                {
+                    _data.ColumnDefinitions = new ColumnDefinitionCollection(this);
+                }
+                _data.ColumnDefinitions.Clear();
+                foreach (var columnDefinition in value)
```

Comment:
This can be done in one go without using the for each loop

Reply:
Done